src/sysupdate/sysupdate.c

```diff
@@ -521,13 +524,32 @@ static int context_show_version(Context *c, const char *version) {
         table_set_ersatz_string(t, TABLE_ERSATZ_DASH);
 
         /* Determine if the target will make use of partition/fs attributes for any of the transfers */
-        for (size_t n = 0; n < c->n_transfers; n++) {
-                Transfer *tr = c->transfers[n];
+        FOREACH_ARRAY(transfer, c->transfers, c->n_transfers) {
+                Transfer *tr = *transfer;
 
                 if (tr->target.type == RESOURCE_PARTITION)
                         show_partition_columns = true;
                 if (RESOURCE_IS_FILESYSTEM(tr->target.type))
                         show_fs_columns = true;
+
+                STRV_FOREACH(changelog, tr->changelog) {
+                        assert(*changelog);
+
+                        char *changelog_url = strreplace(*changelog, "@v", version);
+                        if (!changelog_url)
+                                return log_oom();
+
+                        // Avoid duplicates
+                        if (strv_contains(changelog_urls, changelog_url)) {
+                                free(changelog_url);
+                                continue;
+                        }
+
+                        // changelog_urls takes ownership of expanded changelog_url
+                        r = strv_consume(&changelog_urls, changelog_url);
```

Comment:
and then, when using `_cleanup_free_` use `TAKE_PTR()` on the 2nd argument, so that the auto-destruction is deactivated.